compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.cpp

```diff
@@ -0,0 +1,369 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/RemoveUnnecessaryTransposeNetPass.h"
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/Profile/CircleNodeOrigin.h>
+
+#include <vector>
+
+namespace
+{
+
+class PatternSimulator
```

Comment:
(optional) Better to rename this to something more intuitive, e.g., `TaggedShapeAnalysis`

```suggestion
class PatternSimulator final
```

TMI This kind of analysis is often called "abstract interpretation".